django/utils/dateparse.py

```diff
@@ -156,3 +156,17 @@ def parse_duration(value):
         if match.re == iso8601_duration_re:
             days *= sign
         return days + sign * datetime.timedelta(**kw)
+
+
+def split_timezone_delta(tzname):
+    """
+    Parse delta time in timezone name
+
+    Return name, delta value and sign index.
+    """
+    for sign in ('+', '-'):
+        if sign in tzname:
+            name, offset = tzname.rsplit(sign, 1)
+            if offset and parse_time(offset):
+                return name, f'{sign}{offset}', tzname.find(sign)
+    return tzname, None, -1
```

Comment:
It's probably worth returning `sign` separately, e.g.:
```python
    for sign in ('+', '-'):
        if sign in tzname:
            name, offset = tzname.rsplit(sign, 1)
            if offset and parse_time(offset):
                return name, sign, offset
    return tzname, None, None
```
then we will have a simpler implementation on SQLite:
```python
    if tzname is not None and tzname != conn_tzname:
        tzname, sign, offset = split_timezone_delta(tzname)
        if offset:
            hours, minutes = offset.split(':')
            offset_delta = datetime.timedelta(hours=int(hours), minutes=int(minutes))
            dt += offset_delta if sign == '+' else -offset_delta
```
and PostgreSQL:
```python
        tzname, sign, offset = split_timezone_delta(tzname)
        if offset:
            sign = '-' if sign == '+' else '+'
            return f'{tzname}{sign}{offset}'
        return tzname
```
and maybe a bit more readable on MySQL and Oracle:
```python
tzname, sign, offset = split_timezone_delta(tzname)    
return f'{sign}{offset}' if offset else tzname
```
What do you think?

Reply:
this is definitely better solution. appied it. Thanks. @felixxm